pkg/fleet-manager/application/rollout_helper.go

```diff
@@ -569,6 +705,19 @@ func addLabels(obj client.Object, key, value string) client.Object {
 	return obj
 }
 
+func addAnnotations(obj client.Object, ann map[string]string) client.Object {
+	annotations := obj.GetAnnotations()
+	if annotations == nil {
+		obj.SetAnnotations(ann)
+	} else {
+		for k, v := range ann {
+			annotations[k] = v
```

Comment:
This may change the original annotations, which need to be evaluated to see if there is any .

Reply:
checked